common/functions/src/scalars/others/inet_ntoa.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Datafuse Lfloor.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::str;
+
+use common_datavalues::prelude::DataColumn;
+use common_datavalues::prelude::DataColumnsWithField;
+use common_datavalues::DataSchema;
+use common_datavalues::DataType;
+use common_datavalues::DataValue;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+#[derive(Clone)]
+pub struct InetNtoaFunction {
+    display_name: String,
+}
+
+impl InetNtoaFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(InetNtoaFunction {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic())
+    }
+}
+
+impl Function for InetNtoaFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn num_arguments(&self) -> usize {
+        1
+    }
+
+    fn return_type(&self, args: &[DataType]) -> Result<DataType> {
+        if matches!(args[0], DataType::UInt32) {
+            Ok(DataType::String)
+        } else {
+            Err(ErrorCode::IllegalDataType(format!(
+                "Expected uint32 type, but got {}",
+                args[0]
+            )))
+        }
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(false)
+    }
+
+    fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
+        let ipv4 = columns[0]
+            .column()
+            .to_minimal_array()?
+            .cast_with_type(&DataType::UInt32)?
+            .u32()?
+            .inner()
```

Comment:
If column[0] is not constant, we can't just get the first value.

Let's use resize_constant if it's constant instead.

Reply:
Constant column Example:

select inet_ntoa( toUInt32(123)  ) from numbers(10)

None constant column example:


select inet_ntoa( toUInt32(number)  ) from numbers(10)

Reply:
ok, I change it to `column.resize_constant(columns[0].column().len())`